src/v/redpanda/application.h

```diff
@@ -113,6 +108,12 @@ class application {
     using deferred_actions
       = std::vector<ss::deferred_action<std::function<void()>>>;
 
+    void configure_admin_server();
```

Comment:
I like making these private.  One way to do it without having to put more stuff inside start() would be to `friend` some test fixtures -- I ended up doing that in https://github.com/redpanda-data/redpanda/pull/6529 to enable test fixtures to poke at the state of the feature table.

Not sure how strongly I feel about it though.

Reply:
That is a fair point.  Come to think of it, in that other PR I was adding some "fast forward" cluster setup logic (switch all features on immediately) specifically for the unit tests, that I would really want any fixture to use even if they didn't know the magic.  So it might make sense to have a "start_under_test()" method or similar that wraps everything a fixture should do :thinking:

Reply:
I think I'd prefer something like making the `start()` take some `options` argument to indicate how the caller wants the application to behave, even if it includes test-only args.